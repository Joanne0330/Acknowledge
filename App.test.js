import React from 'react';
import renderer from 'react-test-renderer';
import { act } from 'react-test-renderer';
import App from './App';
import { NavigationContainer } from "@react-navigation/native";


jest.useFakeTimers();


describe('<App />', () => {
  it('has 1 child', async () => {
    const tree = renderer.create(<App />).toJSON();
    await act(async () => { 
      expect(tree.children.length).toBe(1); 
    })
  });

  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
