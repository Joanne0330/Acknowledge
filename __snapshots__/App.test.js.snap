// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransitionStart={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#60DBC5",
                              "elevation": 4,
                              "zIndex": 0,
                            },
                            Object {
                              "shadowColor": "#000000",
                              "shadowOffset": Object {
                                "height": 3,
                                "width": 0,
                              },
                              "shadowOpacity": 0.24,
                              "shadowRadius": 4,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#60DBC5",
                              "elevation": 0,
                              "flexDirection": "row",
                              "height": 56,
                              "marginTop": 0,
                              "paddingHorizontal": 4,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flex": 1,
                                "justifyContent": "center",
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#000000",
                                    "fontFamily": "System",
                                    "fontWeight": "400",
                                    "textAlign": "left",
                                  },
                                  Array [
                                    Object {
                                      "color": "rgba(0, 0, 0, 0.87)",
                                      "fontFamily": "System",
                                      "fontWeight": "500",
                                      "textAlign": "left",
                                      "writingDirection": "ltr",
                                    },
                                    Array [
                                      Object {
                                        "fontSize": 20,
                                        "letterSpacing": 0.15,
                                        "lineHeight": 30,
                                        "marginVertical": 2,
                                      },
                                      Object {
                                        "color": "#2E7166",
                                      },
                                    ],
                                    Array [
                                      Object {
                                        "fontSize": 20,
                                        "letterSpacing": 0.15,
                                        "lineHeight": 30,
                                        "marginVertical": 2,
                                      },
                                      Object {
                                        "color": "#2E7166",
                                      },
                                    ],
                                  ],
                                ]
                              }
                            >
                              Access
                            </Text>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "flex": 1,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "fontSize": 45,
                              "marginBottom": 40,
                              "textAlign": "center",
                            }
                          }
                        >
                          Welcome to Access
                        </Text>
                        <TextInput
                          allowFontScaling={true}
                          autoCapitalize="none"
                          onChangeText={[Function]}
                          placeholder="Email"
                          rejectResponderTermination={true}
                          style={
                            Object {
                              "borderBottomWidth": 1,
                              "borderColor": "#d3d3d3",
                              "fontSize": 16,
                              "margin": 10,
                              "padding": 15,
                              "textAlign": "center",
                              "width": "85%",
                            }
                          }
                          underlineColorAndroid="transparent"
                        />
                        <TextInput
                          allowFontScaling={true}
                          onChangeText={[Function]}
                          placeholder="Password"
                          rejectResponderTermination={true}
                          secureTextEntry={true}
                          style={
                            Object {
                              "borderBottomWidth": 1,
                              "borderColor": "#d3d3d3",
                              "fontSize": 16,
                              "margin": 10,
                              "padding": 15,
                              "textAlign": "center",
                              "width": "85%",
                            }
                          }
                          underlineColorAndroid="transparent"
                        />
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#60DBC5",
                              "borderColor": "#60DBC5",
                              "borderRadius": 5,
                              "borderWidth": 1,
                              "marginBottom": 20,
                              "marginTop": 30,
                              "opacity": 1,
                              "paddingVertical": 5,
                              "width": 300,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#fff",
                                "fontSize": 20,
                                "fontWeight": "bold",
                              }
                            }
                          >
                            Login
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#60DBC5",
                              "borderColor": "#60DBC5",
                              "borderRadius": 5,
                              "borderWidth": 1,
                              "marginBottom": 20,
                              "marginTop": 30,
                              "opacity": 1,
                              "paddingVertical": 5,
                              "width": 300,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#fff",
                                "fontSize": 20,
                                "fontWeight": "bold",
                                "textAlign": "center",
                              }
                            }
                          >
                            Don't have an account yet? 
                            

                             Sign up
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "zIndex": 1,
            },
            Object {
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      />
    </RNCSafeAreaProvider>
  </View>
</View>
`;
